src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsSourceView.cs

```diff
@@ -0,0 +1,312 @@
+		}
+
+		internal void OnItemsSourceChanged(NotifyCollectionChangedEventArgs args)
+		{
+			m_cachedSize = GetSizeCore();
+			CollectionChanged?.Invoke(this, args);
+		}
+
+
+		~InspectingDataSource()
+		{
+			UnListenToCollectionChanges();
+		}
+
+		private int GetSizeCore()
+		{
+			return m_vector.Count;
+		}
+
+		private object GetAtCore(int index)
+		{
+			return m_vector[index];
+		}
+
+		bool HasKeyIndexMappingCore()
+		{
+			return m_uniqueIdMaping != null;
+		}
+
+		string KeyFromIndexCore(int index)
+		{
+			if (m_uniqueIdMaping != null)
+			{
+				return m_uniqueIdMaping.KeyFromIndex(index);
+			}
+			else
+			{
+				throw new InvalidOperationException("ID mapping not set.");
+			}
+		}
+
+		int IndexFromKeyCore(string id)
+		{
+			if (m_uniqueIdMaping != null)
+			{
+				return m_uniqueIdMaping.IndexFromKey(id);
+			}
+			else
+			{
+				throw new InvalidOperationException("ID mapping not set.");
+			}
+		}
+
+		//#pragma endregion
+
+		private int IndexOf(object value)
+		{
+			int index = -1;
+			if (m_vector != null && value != null)
+			{
+				var v = -1;
+				v = m_vector.IndexOf(value);
+				if (v > -1)
+				{
+					index = v;
+				}
+			}
+			return index;
+		}
+
+		//IVector<object> WrapIterable(const IIterable<object>& iterable)
+		//{
+		//	var vector = make < Vector < object, MakeVectorParam< VectorFlag.DependencyObjectBase > () >> ();
+		//	var iterator = iterable.First();
+		//	while (iterator.HasCurrent())
+		//	{
+		//		vector.Append(iterator.Current());
+		//		iterator.MoveNext();
+		//	}
+
+		//	return vector;
+		//}
+
+		//private void UnListenToCollectionChanges()
+		//{
+		//	var notifyCollection = m_notifyCollectionChanged.safe_get()
+		//			if ()
+		//	{
+		//		notifyCollection.CollectionChanged(m_eventToken);
+		//	}
+
+		//	else if (var bindableObservableCollection = m_bindableObservableVector.safe_get())
+		//					{
+		//		bindableObservableCollection.VectorChanged(m_eventToken);
+		//	}
+
+		//					else if (var observableCollection = m_observableVector.safe_get())
+		//					{
+		//		observableCollection.VectorChanged(m_eventToken);
+		//	}
+		//}
+
+		void ListenToCollectionChanges()
+		{
+			if (m_vector == null)
+			{
+				throw new InvalidOperationException("Backing vector not set");
+			}
+			var incc = m_vector as INotifyCollectionChanged;
+			if (incc != null)
+			{
+				m_eventToken = incc.CollectionChanged({ this, &OnCollectionChanged });
+				m_notifyCollectionChanged = incc;
+			}
+			else
+			{
+				var bindableObservableVector = m_vector.try_as<IBindableObservableVector>();
+				if (bindableObservableVector)
+				{
+					m_eventToken = bindableObservableVector.VectorChanged({ this, &OnBindableVectorChanged });
+					m_bindableObservableVector.set(bindableObservableVector);
+				}
+				else
+				{
+					var observableVector = m_vector.try_as<IObservableVector<object>>();
+					if (observableVector)
+					{
+						m_eventToken = observableVector.VectorChanged({ this, &OnVectorChanged });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)